typing/includecore.ml

```diff
@@ -375,8 +383,11 @@ let rec compare_constructor_arguments ~loc env params1 params2 arg1 arg2 =
         Some (Arity : constructor_mismatch)
       else if
         (* Ctype.equal must be called on all arguments at once, cf. PR#7378 *)
-        Ctype.equal env true (params1 @ arg1) (params2 @ arg2)
-      then None else Some Type
+        Ctype.equal env true (params1 @ List.map fst arg1) (params2 @ List.map fst arg2)
+      then
+        (* now check global flags *)
+        Option.map (fun err -> Nonlocality err) (List.find_map (fun (x,y) -> compare_global_flags x y) (List.combine (List.map snd arg1) (List.map snd arg2)))
```

Comment:
It would be clearer if the steps were separated out:
```ocaml
  then begin
    let gfs = List.combine arg_gfs1 arg_gfs2 in
    let err = List.find_map (fun (gf1, gf2) -> compare_global_flags gf1 gf2) gfs in
    Option.map (fun e -> Nonlocality e) err
  end
  ```
  or if you prefer:
  ```ocaml
  then
     List.combine arg_gfs1 arg_gfs2
     |> List.find_map (fun (gf1, gf2) -> compare_global_flags gf1 gf2)
     |> Option.map (fun e -> Nonlocality e)
  ```